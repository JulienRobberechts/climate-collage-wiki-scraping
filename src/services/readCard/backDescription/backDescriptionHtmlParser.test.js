const { getContent } = require("../../utils/fileServices/readFile");
const { parseBackDescription } = require("./backDescriptionHtmlParser");

// to update those files check manual-calls/get -card-sections.http
const cardContent1FilePath = "./src/tests/data/fr/carte1-first-section.html";
const cardContent3FilePath = "./src/tests/data/fr/carte3-first-section.html";

describe("parse BackDescription", () => {
  it("parse BackDescription card 1", async () => {
    const content = await getContent(cardContent1FilePath);
    const result = parseBackDescription(content);
    expect(result).toContain("que tout commence");
  });
  it("parse BackDescription card 3", async () => {
    const content = await getContent(cardContent3FilePath);
    const result = parseBackDescription(content);
    expect(result).toMatchSnapshot();
  });
});
