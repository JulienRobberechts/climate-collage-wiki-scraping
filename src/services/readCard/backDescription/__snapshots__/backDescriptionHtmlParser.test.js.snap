// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse BackDescription parse BackDescription card 3 1`] = `
"\\\\nL\\\\u2019utilisation du b\\\\u00e2timent (logement et services) utilise des \\\\u00e9nergies fossiles et de
      l'\\\\u00e9lectricit\\\\u00e9.\\\\nCela repr\\\\u00e9sente 20% des Gaz \\\\u00e0 Effet de Serre (GES).\\\\n"
`;
